flask_resources/resources.py

```diff
@@ -33,14 +33,44 @@ class ResourceConfig:
     item_route = "/resources/<id>"
     list_route = "/resources/"
 
+    # Suggested
+    # incoming = {
+    #     "item_route": "/resources/<id>",
+    #     "list_route": "/resources/",
+
+    #     "args": Parser(),
+    #     # or
+    #     "args": {
+    #         "create": Parser()  # any of the resource method can be listed
+    #     },
+    #     # both are possible (One-for-all 1st)
+    #     "body": [JSONDeserializer(mimetype="application/json"),],
+    #     # or
+    #     "body": {
+    #         "create": [JSONDeserializer(mimetype="application/json"),]
+    #     }
+    #     # both are possible
+    # }
+
+    # outgoing = {
+    #     "headers": MakeHeaderFunction(override=False)  # Optional
```

Comment:
While working on the response serializing for "invenio". Meaning, link injection, add total results, add aggregations... If we do not define (or at least give the option to) the `Response` class we will simply have to overwrite the full configuration in Invenio-Resources (sort of contradicts the purpose...).

Same happens with the loader. Even if not needed in configuration here. If it is a different one (for reason X) where would it be configured/set?

Reply:
This will be clearer in the upcoming PR, but the approach is:

`Response` splits into `headers` and `serializers` configurations. To inject links, you configure headers and serializers to you custom serializers. We could have some nicer ones to inherit from, but that's sugar to me ;) .
 
Loaders should already be taken care by deserializers. 

code will make this clearer.

Reply:
In this PR I address request parsing + `default_config` . In the next PR I will address this refactor of configuration.